In [2]:
import sys
sys.path.append('../')
import utils.data_preprocess as preprocess

In [3]:
pre = preprocess.preprocess("D:/DataSet/CICIDS2017/CSV/Wednesday-workingHours.pcap_ISCX.csv", labelColumnName='Label', depth=2)
X_train, X_test, y_train, y_test = pre.preprocessing()

In [27]:
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression( solver='saga')
classifier.fit(X_train, y_train.argmax(axis=1))  # y_train = 0 is attack

LogisticRegression(solver='saga')

In [28]:
y_pred = classifier.predict(X_test)

In [29]:
from sklearn.metrics import confusion_matrix
confusion_matrix(y_test.argmax(axis=1), y_pred)

array([[125821,   6124],
       [  3317,  72160]], dtype=int64)

In [30]:
from sklearn import metrics
print("F1 Score : ", metrics.f1_score(y_test.argmax(axis=1), y_pred, average='binary'))
print("Precision : ", metrics.precision_score(y_test.argmax(axis=1), y_pred, average='binary'))
print("Recall : ", metrics.recall_score(y_test.argmax(axis=1), y_pred, average='binary'))

F1 Score :  0.9385995148314592
Precision :  0.9217720096060498
Recall :  0.9560528372881805


### lbfgs
```
array([[126181,   5764],
       [  2632,  72845]], dtype=int64)
```
F1 Score :  0.9455109484313955
Precision :  0.9266750626518592
Recall :  0.9651284497264068
### saga
```
array([[125821,   6124],
       [  3317,  72160]], dtype=int64)
```
F1 Score :  0.9385995148314592
Precision :  0.9217720096060498
Recall :  0.9560528372881805